aws/kinesis/core/pipeline.h

```diff
@@ -190,8 +200,34 @@ class Pipeline : boost::noncopyable {
     });
   }
 
-  std::string stream_;
+  // Retrieve the account ID and partition from the STS service.
+  static std::string init_stream_arn(const std::string &region, const std::string &stream_name) {
+    Aws::STS::STSClient sts;
+    Aws::STS::Model::GetCallerIdentityRequest request;
+    auto outcome = sts.GetCallerIdentity(request);
+    if (outcome.IsSuccess()) {
+      auto result = outcome.GetResult();
+      Aws::Utils::ARN sts_arn(result.GetArn());
+
+      // Construct and return the Kinesis stream ARN.
+      std::stringstream arn;
+      arn << "arn:" << sts_arn.GetPartition() << ":kinesis:" << region << ":" << result.GetAccount()
+          << ":stream/" << stream_name;
```

Comment:
Is this stream arn format same for all regions? I thought we had some issues with arn format in some regions?

Reply:
That's a good call out, we once had a problem with pod1, but I think the format is good for all public commercial regions

Reply:
yep, we should still test to make sure this works in all partitions.